File: dontGiveMeFiveReally/kata.js


function dontGiveMeFive(start, end) {
    let counter = 0;
    // get 'start' to a number containing five
    while (!start.toString().includes('5')) {
        start++;
        if (!start.toString().includes('5'))
            counter++;
    }

    for (let index = start; index <= end; index += 5) {
        console.log(index);
        counter += 4;
        if (!index.toString().includes('5'))
            counter++;
        console.log('couynter = ' + counter);
        if (index + 5 > end) {
            for (let j = index; j <= end; j++) {
                console.log('j = ' + j);

                if (!j.toString().includes('5'))
                    counter++;

            }
            break;
        }
    }
    return counter;
}

console.log('value = ' + dontGiveMeFive(4, 17));

File: dontGiveMeFiveReally/kata.py

import math


def dont_give_me_five(start, end):
    nonFiveCounter = 0
    for i in range(start, end+1):
        nonFiveCounter += containsFive(i)
    
    return nonFiveCounter

def containsFive(number):
    while(number != 0):
        if (number % 10 == 5):
            return 0
        number = int(number/10)
    return 1

print(dont_give_me_five(-9000000000000000000, 9000000000000000000))